File: WebApi/Models/Maintenance/Maintenace.cs
﻿using System.ComponentModel.DataAnnotations;
using WebApi.Models.Users;
using WebApi.Models.Property;
namespace WebApi.Models.Maintenance
{
    public class Maintenance
    {
        [Key]
        public int MaintenanceID { get; set; }
        public int PropertyID { get; set; }
        public int? ContractorID { get; set; }
        public int? MaintenanceStatusID { get; set; }
        public int? MaintenanceTypeID { get; set; }
        public DateTime? MaintenanceDate { get; set; }
        public string? MaintenanceTime { get; set; }
        public Property.Property Property { get; set; }
        public Contractor Contractor { get; set; }
        public MaintenanceStatus MaintenanceStatus { get; set; }
        public MaintenanceType MaintenanceType { get; set; }    

    }
}

public class MaintenanceUpdateModel
{
    public int ContractorID { get; set; }
    public DateTime MaintenanceDate { get; set; }
    public string MaintenanceTime { get; set; }
    public int MaintenanceTypeID { get; set; }
}

File: tenant-app/src/app/shared/Maintenace.ts
import { Contractor } from "./UserModels/Contractor";
import { MaintenanceStatus } from "./MaintenanceStatus";
import { Property } from "./Property/Property";
import { MaintenanceType } from "./MaintenanceType";

export class Maintenance{
  propertyID: number=0;
  contractorID: number=0;
  maintenanceStatusID: any;
  maintenanceTypeID: number=0;
  maintenanceDate: string = "";
  maintenanceTime:string = "";
  property: Property=new Property;
  contractor: Contractor =new Contractor;
  maintenanceStatus:MaintenanceStatus=new MaintenanceStatus;
  maintenanceType:MaintenanceType=new MaintenanceType;

  constructor(){
  }
}
